examples/rl/rllib/rllib_agent.py

```diff
@@ -41,19 +42,38 @@
     }
 )
 
+FLATTENED_OBSERVATION_SPACE = gym.spaces.utils.flatten_space(OBSERVATION_SPACE)
 
-def observation_adapter(env_observation):
-    return lane_ttc_observation_adapter.transform(env_observation)
 
+def observation_adapter(agent_observation, /):
+    return gym.spaces.utils.flatten(
+        OBSERVATION_SPACE, lane_ttc_observation_adapter.transform(agent_observation)
+    )
 
-def action_adapter(model_action):
-    throttle, brake, steering = model_action
-    return np.array([throttle, brake, steering * np.pi * 0.25])
+
+def action_adapter(agent_action, /):
+    throttle, brake, steering = agent_action
+    return np.array([throttle, brake, steering * np.pi * 0.25], dtype=np.float32)
 
 
 class TrainingModel(FullyConnectedNetwork):
     NAME = "FullyConnectedNetwork"
 
+    def __init__(
+        self,
+        obs_space,
+        action_space,
+        num_outputs: int,
+        model_config: ModelConfigDict,
+        name: str,
+    ):
+        super().__init__(obs_space, action_space, num_outputs, model_config, name)
+
+    def forward(self, input_dict, state, seq_lens):
```

Comment:
One of the issues here is that the observation and action spaces are expected to be formed at this point (these are given at the start).

I could inject the action and observation space adaptors in here but this is already after the tensors have been generated... I am unsure if there is a way to intercept them. I found an experimental `observation_fn` but it appears it does not inject between the observation and the policy.

We may have to keep the action space and observation space adaptors for `RLlibHiWayEnv`.